src/test/java/tests/TestUtils.java

```diff
@@ -129,4 +130,28 @@ public static HttpRequest createMockServerRequest(String method, int page,
                 .withQueryStringParameter("per_page", "100")
                 .withQueryStringParameter("page", Integer.toString(page));
     }
+
+    /**
+     * Delays for a specified time then applies a function and returns its result
+     * @param delay time to be delayed in milliseconds
+     * @param supplier
+     * @return
+     * @throws InterruptedException
+     */
+    public static <T> T delayThenApply(long delay, Supplier<T> supplier) throws InterruptedException {
```

Comment:
`apply` is the name of the method for `Function`. Maybe use `delayThenGet` here?